releasenotes/notes/improve-register-length-handling-cb55f644da136828.yaml

```diff
@@ -0,0 +1,7 @@
+---
+features:
+  - |
+    The :attr:`.SparseLabelOp.register_length` attribute (and by extension that of its subclasses, too)
+    can no longer take ``None`` as its value. However, the attributes which this one might rely on
+    (e.g. :attr:`.FermionicOp.num_spin_orbitals` or :attr:`.BosonicOp.num_modes`) can remain ``None``.
+    This ensures that the lower-bound behavior works as intended.
```

Comment:
I see the Attributes: docstring for these in the two classes for some reason do not have the type in them - now it would be int | None I should think. It would be good to add the typehint.

Do you imagine any cases where users would need to be wary with existing code - if they relied on getting a value back before with None now I imagine code could have problems. We could add an upgrade section and call this aspect out to say users might need to upgrade code using these attributes to now cater to None?

Reply:
Actually, I do not think the previous code every truely return `None` :thinking: 
Unless they used `copy=False` but then other cases would likely have been broken for them.

I will update the type hints though